src/Symfony/Component/Lock/Store/MemcachedStore.php

```diff
@@ -0,0 +1,179 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Lock\Store;
+
+use Symfony\Component\Lock\Exception\InvalidArgumentException;
+use Symfony\Component\Lock\Exception\LockConflictedException;
+use Symfony\Component\Lock\Key;
+use Symfony\Component\Lock\StoreInterface;
+
+/**
+ * MemcachedStore is a StoreInterface implementation using Memcached as store engine.
+ *
+ * @author Jérémy Derussé <jeremy@derusse.com>
+ */
+class MemcachedStore implements StoreInterface
+{
+    private $memcached;
+    private $initialTtl;
+    /** @var bool */
+    private $useExtendedReturn;
+
+    public static function isSupported()
+    {
+        return extension_loaded('memcached');
+    }
+
+    /**
+     * @param \Memcached $memcached
+     * @param int        $initialTtl the expiration delay of locks in seconds
+     */
+    public function __construct(\Memcached $memcached, $initialTtl = 300)
+    {
+        if (!static::isSupported()) {
```

Comment:
is it possible to have an object instance of  `\Memcached`  and not having the `memcached` extension loaded?